### DISCUSSION TITLE
LDAP

### DISCUSSION THREAD

**Discussion by cyberb** [dani-garcia/vaultwarden#1539] [Ideas]
I wan to implement a simple LDAP auth as I need it for Syncloud (https://github.com/syncloud/platform) project.
Similar to this (https://github.com/cyberb/api/commit/2d011fff7df062d36a6044806d2b7f893e74e334) once done for a rust pi-hole fork.
I have quickly checked the code and I think the logic will be as follows:

1. Configure ldap (enabled, address, binddn)
2. Change webui to allow non email username (optional)
3. Backend: if enabled and authenticated against LDAP
4. Backend: register user with the info from LDAP record if does not exist yet.

Any advices before I start?
I will do this anyway even if it will stay in my fork.

**Comment by nightmared** [comment]
Hello, I'm interested in such a feature too, but I wonder how this can work given the problems discussed at https://github.com/dani-garcia/bitwarden_rs/pull/677#issuecomment-553606347 (the fact that a password change in the LDAP directory will make the vault inacessible). Do you plan on adding an ldap attribute to store t fixed passwordfor the vault, or do you have another idea to solve this hurdle?

Thanks!

**Comment by BlackDex** [comment]
There already is a working solution which uses the official support directory tools from bitwarden. See:  https://github.com/dani-garcia/bitwarden_rs/commit/85e3c73525d327042c1ad142e48c044a5dbdd89c
LDAP and SSO can only be used for account creation or validation. Not for auto login. You always need a master password to decrypt the vault.
So, what would be an addition to this project would be SSO, but that is a huge endeavor.

**Reply by mvalois** [reply]
@BlackDex How can I contact you to work on this together?

**Reply by BlackDex** [reply]
I would suggest to ask @Sheap since he is already working on an SSO implementation.
See #1955

**Comment by cyberb** [comment]
Sory how can I miss that all passwords are also encrypted on the server side using master password.
It sounds crazy, but is it a big security problem to not encrypt passwords in self-hosting mode?
Am I protecting from someone breaking into my home server and not seeing my mail password while having all my mail unencrypted in a separate dir as mail server runs on the same box?

**Reply by BlackDex** [reply]
Not sure what you mean, but passwords are encrypted. All is encrypted, except name and email of the user. Org names ar also not encrypted. But that is about it. The rest is all encrypted.

**Comment by cyberb** [comment]
I mean is it possible to disable encryption?

**Reply by BlackDex** [reply]
Nope. As all is done by the clients. They encrypt and decrypt the data. Also, i would never ever use a password manager which stores my passwords unencrypted and only used a password for login validation.
